### PR TITLE
Fix the endian bug in PauliSumOp

### PR DESCRIPTION
**Pull request by ikkoham**
### Summary

I misunderstood about the endian in the PauliSumOp.
I think this endian is correct, but please double check it.

[test_vqe.py](https://github.com/Qiskit/qiskit-terra/compare/master...ikkoham:fix-paulisumop-eval?expand=1#diff-5d988c768c31118bf11473a07191e32dbb3a7dfde87f5631c7b34fb7d0e602d6R150) is the revert of the previous PR https://github.com/Qiskit/qiskit-terra/pull/5623.

### CODE REVIEW

File: test/python/algorithms/test_vqe.py [hunk 1]
```diff
@@ -147,7 +147,7 @@ def test_basic_aer_qasm(self):
 
         # TODO benchmark this later.
         result = vqe.compute_minimum_eigenvalue(operator=self.h2_op)
-        self.assertAlmostEqual(result.eigenvalue.real, self.h2_energy, places=3)
+        self.assertAlmostEqual(result.eigenvalue.real, -1.86823, places=2)
```
Comment (t-imamichi) [inline test/python/algorithms/test_vqe.py:150]: Why does it not converge to `self.h2_energy`? Do you need to change the options of SPSA?

Comment (ikkoham) [reply test/python/algorithms/test_vqe.py:150]: I am not sure... But I found it depends on the random seed and the number of measurements (1024 now).

Comment (ikkoham) [reply test/python/algorithms/test_vqe.py:150]: As I wrote the Pull Request message, this was original in opflow.

Comment (woodsp-ibm) [reply test/python/algorithms/test_vqe.py:150]: Yes, it does depend on seeds and shots. If it is what it was before a recent change then it seems fine. As you can see its actually found a value below the true value due to the sampling noise.

**Review by woodsp-ibm — APPROVED** [review]

**Pull request merged**